mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/ConvertScreenV2.tsx

```diff
@@ -0,0 +1,355 @@
+        light={tailwind('text-mono-light-v2-700')} dark={tailwind('text-mono-dark-v2-700')}
+      >
+        {`${translate('screens/ConvertScreen', props.targetUnit)})`}
+      </ThemedTextV2>
+    </View>
+  )
+}
+
+function ConversionResultCard (props: { unit: string | undefined, oriTargetAmount: string, totalTargetAmount: string }): JSX.Element {
+  return (
+    <ThemedViewV2
+      style={tailwind('flex-col w-full p-5 mt-6 rounded-lg-v2 border-0.5')} testID='convert_result_card'
+      light={tailwind('border-mono-light-v2-300')} dark={tailwind('border-mono-dark-v2-300')}
+    >
+      <ThemedViewV2
+        style={tailwind('flex-row items-center pb-5')}
+      >
+        <ThemedTextV2
+          style={tailwind('font-normal-v2 text-sm pr-2')} testID='convert_available_label'
+          light={tailwind('text-mono-light-v2-500')} dark={tailwind('text-mono-dark-v2-500')}
+        >
+          {`${translate('screens/ConvertScreen', 'Available {{unit}}', { unit: props.unit })}`}
+        </ThemedTextV2>
+        <ThemedTextV2
+          style={tailwind('flex-1 font-normal-v2 text-sm text-right')} testID='convert_available_amount'
+          light={tailwind('text-mono-light-v2-800')} dark={tailwind('text-mono-dark-v2-800')}
+        >
+          {new BigNumber(props.oriTargetAmount).toFixed(8)}
+        </ThemedTextV2>
+      </ThemedViewV2>
+      <ThemedViewV2
+        style={tailwind('flex-row items-center pt-5 border-t-0.5')}
+        light={tailwind('border-mono-light-v2-300')} dark={tailwind('border-mono-dark-v2-300')}
+      >
+        <ThemedTextV2
+          style={tailwind('font-normal-v2 text-sm pr-2')} testID='convert_resulting_label'
+          light={tailwind('text-mono-light-v2-500')} dark={tailwind('text-mono-dark-v2-500')}
+        >
+          {`${translate('screens/ConvertScreen', 'Resulting {{unit}}', { unit: props.unit })}`}
+        </ThemedTextV2>
+        <ThemedTextV2
+          style={tailwind('flex-1 font-semibold-v2 text-sm text-right')} testID='convert_result_amount'
+          light={tailwind('text-mono-light-v2-800')} dark={tailwind('text-mono-dark-v2-800')}
+        >
+          {props.totalTargetAmount}
+        </ThemedTextV2>
+      </ThemedViewV2>
+    </ThemedViewV2>
+  )
+}
+
+function canConvert (amount: string, balance: string): boolean {
+  return new BigNumber(balance).gte(amount) && !(new BigNumber(amount).isZero()) && (new BigNumber(amount).isPositive())
+}
+
+function getConvertibleUtxoAmount (mode: ConversionMode, source: AddressToken): string {
+  if (mode === 'accountToUtxos') {
+    return source.amount
+  }
+
+  const utxoToReserve = '0.1'
+  const leftover = new BigNumber(source.amount).minus(new BigNumber(utxoToReserve))
+  return leftover.isLessThan(0) ? '0' : leftover.toFixed()
+}
+
+function isUtxoToAccount (mode: ConversionMode): boolean {
+  return mode === 'utxosToAccount'
+}
+
+function getDisplayUnit (unit?: string): string | undefined {
```

Comment:
To be more explicit, and to be re-use by `ConvertConfirmationScreenV2`
```suggestion
export function getDisplayUnit (unit: 'UTXO' | 'Token'): 'UTXO' | 'tokens' {
```

Reply:
done, updated to more explicit and reusable